storage/storage_reference.go

```diff
@@ -0,0 +1,96 @@
+package storage
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/types"
+	"github.com/containers/storage/storage"
+)
+
+// A storageReference holds a name and/or an ID, which is a 32-byte value
+// hex-encoded into a 64-character string.
+type storageReference struct {
+	store     storage.Store
+	transport *storageTransport
+	reference string
+	id        string
+}
+
+func newReference(store storage.Store, transport *storageTransport, reference, id string) *storageReference {
+	return &storageReference{
+		store:     store,
+		transport: transport,
+		reference: reference,
+		id:        id,
+	}
+}
+
+// Resolve the reference's name to an image ID in the storage library if
+// there's already one present with the same name or ID.
+func (s *storageReference) ID() string {
```

Comment:
I once thought that it would need to be, but nothing else ends up using it.  Renaming it to resolveID().